scripts/entity/components/movement_components/player_movement_component.gd

```diff
@@ -42,17 +51,25 @@ func _ready() -> void:
 ## Applies movement to the character.
 func _physics_process(delta: float) -> void:
 	super._physics_process(delta)
-	apply_gravity(delta)
-	apply_horizontal_movement(delta)
+	if _is_climbing:
+		apply_climbing_movement(delta)
+	else:
+		apply_gravity(delta)
+		apply_horizontal_movement(delta)
 	move_and_slide()
 
-func on_movement_input(horizontal_vector2: float) -> void:
-	_current_input_direction = horizontal_vector2
+func on_horizontal_movement_input(horizontal_vector2: float) -> void:
+	_current_input_direction.x = horizontal_vector2
+
+func on_vertical_movement_input(vertical_vector2: float) -> void:
+	_current_input_direction.y = vertical_vector2
+	_is_up_pressed = vertical_vector2 < 0
 
 func on_jump_input_started() -> void:
-	if is_on_floor():
+	if is_on_floor() or _is_climbing:
 		_can_moth_jump = true
-		velocity.y = JUMP_VELOCITY
+		velocity.y = LAUNCH_VELOCITY if _is_climbing else JUMP_VELOCITY
+		_is_climbing = false
```

Comment:
why cant I climb the ladder I am in after I jump?

Reply:
Yeah idk, fixed that